ui/pages/users/index.js

```diff
@@ -80,113 +93,91 @@ function Details({ user, admin, onDelete }) {
+                  const usersToRemove = [id]
+                  await fetch(`/api/groups/${groupId}/users`, {
+                    method: 'PATCH',
+                    body: JSON.stringify({ usersToRemove }),
                   })
-
-                  // delete old grant
-                  await fetch(`/api/grants/${g.id}`, { method: 'DELETE' })
-
-                  mutate({
-                    items: [
-                      ...grants.filter(f => f.id !== g.id),
-                      await res.json(),
-                    ],
+                  mutateGroups({
+                    items: groups.filter(i => i.id !== groupId),
                   })
                 }}
               />
             </div>
-          ))}
-          {inherited?.sort(sortByResource)?.map(g => (
-            <div
-              key={g.id}
-              className='flex items-center justify-between text-2xs'
-            >
-              <div>{g.resource}</div>
-              <div className='flex flex-none'>
-                <div
-                  title='This access is inherited by a group and cannot be edited here'
-                  className='relative mx-1 self-center rounded border border-gray-800 bg-gray-800 px-2 pt-px text-2xs text-gray-400'
-                >
-                  inherited
-                </div>
-                <div className='relative w-32 flex-none py-2 pl-3 pr-8 text-left text-2xs text-gray-400'>
-                  {g.privilege}
-                </div>
-              </div>
-            </div>
-          ))}
-          {!grants?.length && !inherited?.length && !loading && (
-            <div className='mt-6 text-2xs italic text-gray-400'>No access</div>
-          )}
-        </section>
+          </section>
+        </>
       )}
       <section>
         <h3 className='border-b border-gray-800 py-4 text-3xs uppercase text-gray-400'>
           Metadata
         </h3>
-        <div className='flex flex-col space-y-2 pt-3'>
-          <div className='flex flex-row items-center'>
-            <div className='w-1/3 text-2xs text-gray-400'>ID</div>
-            <div className='text-2xs'>{user?.id}</div>
-          </div>
-          <div className='flex flex-row items-center'>
-            <div className='w-1/3 text-2xs text-gray-400'>Created</div>
-            <div className='text-2xs'>
-              {user?.created ? dayjs(user.created).fromNow() : '-'}
-            </div>
-          </div>
-        </div>
+        <Metadata data={metadata} />
       </section>
       <section className='flex flex-1 flex-col items-end justify-end py-6'>
         {auth.id !== id && (
-          <button
-            type='button'
-            onClick={() => setDeleteModalOpen(true)}
-            className='flex items-center rounded-md border border-violet-300 px-6 py-3 text-2xs text-violet-100'
-          >
-            Remove
-          </button>
+          <RemoveButton
+            modalOpen={deleteModalOpen}
+            setModalOpen={setDeleteModalOpen}
```

Comment:
In this case the parent component doesn't really need to hold the state for if the modal is open or not: we can remove these props (unless they are used elsewhere) for the `RemoveButton` component